packages/protocol/contracts-0.8/common/EpochManager.sol

```diff
@@ -506,4 +649,82 @@ contract EpochManager is
       CELOequivalent
     );
   }
+
+  /**
+   * @notice Updates the list of elected validator signers.
+   */
+  function _setElectedSigners(address[] memory _elected) internal {
+    require(electedAccounts.length > 0, "Elected list length cannot be zero.");
+    IAccounts accounts = getAccounts();
+    electedSigners = new address[](_elected.length);
+    for (uint i = 0; i < _elected.length; i++) {
+      electedSigners[i] = accounts.getValidatorSigner(_elected[i]);
+    }
+  }
+
+  /**
+   * @notice Returns the epoch info of a specified blockNumber.
+   * @param _blockNumber Block number of the epoch info is retreived.
+   * @return firstEpoch The first block of the given block number.
+   * @return lastBlock The first block of the given block number.
+   * @return startTimestamp The starting timestamp of the given block number.
+   * @return rewardsBlock The reward block of the given block number.
+   */
+  function _getEpochByBlockNumber(
```

Comment:
This function is very likely to run out of gas, is it (and the functions that use it) really needed?

I also think that it has quite some edge cases, would be difficult to test them all.

Reply:
comment added